### ISSUE TITLE
toBase64Image

### ISSUE THREAD

**Issue by singhqwerty6** [jerairrest/react-chartjs-2#464]
Hi, I am new to React .

Chart.js instance
chartReference = {};

componentDidMount() {
  console.log(this.chartReference); // returns a Chart.js instance reference
}

render() {
  return (<Doughnut ref={(reference) => this.chartReference = reference } data={data} />)
}

how to use this.chartReference to get a base 64 encoded string of the chart in it's current state.
I tried this.chartReference.toBase64Image but it is not function error

**Comment by singhqwerty6** [comment]
Got it

**Comment by Susmitha-Bogala** [comment]
could you please explain, how to convert chart into base64 format?

**Comment by singhqwerty6** [comment]
create a ref and use its instant

**Comment by Susmitha-Bogala** [comment]
i tried to do the same.but getting "toBase64Image is not a function" this error.

**Comment by simromm** [comment]
Same problem here, I create a ref and use it. But"toBase64Image is not a function"

**Comment by singhqwerty6** [comment]
Console log that instant ref and search for ref.chartinstance.base64

**Comment by simromm** [comment]
I already did but nothing founded.
I use also this method => https://github.com/jerairrest/react-chartjs-2#chartjs-instance
to get mychartjs instance.

Then console.log(this.chartReference) when the component is mounted.

![ref](https://user-images.githubusercontent.com/48606574/76420923-2e8c7f00-63a3-11ea-8233-65f795932b74.png)

**Comment by singhqwerty6** [comment]
Email me personally. I ll look into it.

**Comment by jdgarcia1381** [comment]
You need to get a ChartJS instance reference and then call `.toBase64Image()`. To do that, first create an empty reference:

    const chartRef = useRef(null);

Then, assing the reference. For example, if you are using a `Bar` chart:

    <Bar data={data} ref={chartRef}/>

Finally, get the base64 image using the reference. **Remember to call `.current.chartInstance` before `toBase64Image()`**:

    const base64Image = chartRef.current.chartInstance.toBase64Image();

**Comment by prakashsubedi0440** [comment]
I think, at this point of time, react-chart-js has migrated. So, for me, when I did const base64Image = chartRef.current.toBase64Image() ; instead of const base64Image = chartRef.current.chartInstance.toBase64Image(), I could get the values.

But still I when I saved in my harddisk (using react-download-link), and tried to open , it said, file format not supported. I could only open in txt version, but not image
